llvm/include/llvm/Analysis/DXILResource.h

```diff
@@ -353,13 +340,22 @@ class ResourceInfo {
   GlobalVariable *Symbol = nullptr;
 
 public:
+  bool GloballyCoherent = false;
+  ResourceCounterDirection CounterDirection = ResourceCounterDirection::Unknown;
```

Comment:
I'm surprised to see public member variables as well as setter functions for them.

Reply:
Is there a preference for one vs the other? The members need to be read and written externally so I can make them private and add getters or just delete the setters.

FWIW the previous location for these members (I'm just moving them) had public+setters though I don't think the setters were used

Reply:
I'll have to defer to @llvm-beanz or @bogner here I think.  I have opinions but I'm not sure they're aligned to LLVM's best practices.

Reply:
picked the one with least churn so we can get the PR in. If someone comes along later to disagree, I'll happily create a new PR to swap them!